libvast/include/vast/system/builtin_rest_endpoints.hpp

```diff
@@ -0,0 +1,21 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <cstdint>
+
+namespace vast::system {
+
+// All endpoint ids of the built-in VAST endpoints.
+
+enum class query_endpoints : uint64_t { new_, next };
+
+enum class status_endpoints : uint64_t {
+  status,
+};
```

Comment:
Ideally this would be in a header file belonging to the plugin itself, but I'm not sure we can do that with our current scaffolding for builtin plugins?

Reply:
It's not possible, but can we proxy without needing these IDs instead? This looks rather brittle. When I want to proxy "/query/new", shouldn't it suffice to say that I want to proxy "/query/new" instead of saying that I want to proxy "/query" with the endpoint ID `query_endpoints::new_`?

Reply:
The underlying idea is that we don't want to duplicate the routing logic in the server to decide which path corresponds to which endpoint. And this is not trivial, since you can have the same path for different endpoints (ie. GET, POST), and can have endpoints with non-static paths (`/query/6c7bda29-67bc-48b5-9167-cc34f5b4c640/next`).

One approach to avoid this could be to add a method to build a unique canonical path (that looks like `GET /query/:id/next` from an endpoint and then string-match based on that.